src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,57 @@
-# TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
+import random
+
+# TO-DO: Complete the selection_sort() function below
+
+
+def selection_sort(arr):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
+        # (hint, can do in 3 loc)
+        for j in range(i, len(arr)):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
 
         # TO-DO: swap
-
-
-
+        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
 
-    return arr
 
+def bubble_sort(arr):
+    swapped = True
+    while swapped:
+        swapped = False
+        for i in range(1, len(arr)):
+            if arr[i] < arr[i - 1]:
+                arr[i], arr[i-1] = arr[i-1], arr[i]
+                swapped = True
+
+    return arr
 
 # STRETCH: implement the Count Sort function below
-def count_sort( arr, maximum=-1 ):
 
-    return arr
\ No newline at end of file
+
+def count_sort(arr, maximum=-1):
+    counts = []
+    output = []
+    if len(arr) == 0:
+        return arr
+    if maximum == -1:
+        maximum = max(arr)
+    for i in range(0, maximum+1):
```

Comment:
To create an array with the length of the maximum you can also do `counts = [0] * maximum`